lib/torrent.js

```diff
@@ -568,10 +575,18 @@ class Torrent extends EventEmitter {
           cb(null)
         })
       })
-    }), FILESYSTEM_CONCURRENCY, err => {
-      if (err) return this._destroy(err)
-      this._debug('done verifying')
-      this._onStore()
+    }), FILESYSTEM_CONCURRENCY, cb)
+  }
+
+  rescanFiles (cb) {
+    if (!cb) cb = noop
+
+    // already checks for torrent destroyed
+    this._verifyPieces((err) => {
+      if (err) return cb(err)
```

Comment:
In addition to calling the user's callback, shouldn't this also call `this._destroy(err)`? That is what you do in the other case where `_verifyPieces` returns an error.

Reply:
Yes it should.